src/BinSkim.Driver/AnalysisSummaryExtractor.cs

```diff
@@ -51,17 +57,44 @@ public static void UpdateBuildPipelineInfo(AnalysisSummary summary)
             // https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
             if (summary != null)
             {
-                try
-                {
-                    summary.BuildDefinitionId = Environment.GetEnvironmentVariable(BuildDefinitionIdVar);
-                    summary.BuildDefinitionName = Environment.GetEnvironmentVariable(BuildDefinitionNameVar);
-                    summary.BuildRunId = Environment.GetEnvironmentVariable(BuildDefinitionRunIdVar);
-                }
-                catch (SecurityException)
+                summary.BuildDefinitionId = ExtractValueFromEnvironmentVariable(BuildDefinitionIdVar);
+                summary.BuildDefinitionName = ExtractValueFromEnvironmentVariable(BuildDefinitionNameVar);
+                summary.BuildRunId = ExtractValueFromEnvironmentVariable(BuildDefinitionRunIdVar);
+
+                summary.RepositoryId = ExtractValueFromEnvironmentVariable(RepositoryIdVar);
+                summary.RepositoryName = ExtractValueFromEnvironmentVariable(RepositoryNameVar);
+
+                summary.ProjectId = ExtractValueFromEnvironmentVariable(ProjectIdVar);
+                summary.ProjectName = ExtractValueFromEnvironmentVariable(ProjectNameVar);
+
+                summary.OrganizationId = ExtractValueFromEnvironmentVariable(OrganizationIdVar);
+                summary.OrganizationName = ExtractValueFromEnvironmentVariable(OrganizationNameVar);
+                summary.OrganizationName = summary.OrganizationName.Replace("https://dev.azure.com/", string.Empty).TrimEnd('/');
```

Comment:
>summary.OrganizationName [](http://example.com/codeflow?start=43&length=24)

the logic in the method ExtractValueFromEnvironmentVariable() looks like if can not find will return null, so this can be null 
maybe add a "?" here

Reply:
changed the method that is called to always output an empty string if no envvariable  was found.
that will prevent us to add `?` everywhere.

Reply:
good idea!